src/Microsoft.TemplateEngine.Cli/TemplateResolution/ListOrHelpTemplateListResolutionResult.cs

```diff
@@ -0,0 +1,241 @@
+        /// Returns true when at least one template has mismatch in context (type)
+        /// </summary>
+        public bool HasContextMismatch
+        {
+            get
+            {
+                return PartiallyMatchedTemplates.Any(t => t.HasContextMismatch());
+            }
+        }
+
+        /// <summary>
+        /// Returns true when at least one template has mismatch in baseline
+        /// </summary>
+        public bool HasBaselineMismatch
+        {
+            get
+            {
+                return PartiallyMatchedTemplates.Any(t => t.HasBaselineMismatch());
+            }
+        }
+
+        /// <summary>
+        /// Returns true when one and only one template has exact match
+        /// </summary>
+        public bool HasUnambiguousTemplateGroup
+        {
+            get
+            {
+                return ExactMatchedTemplatesGrouped.Count == 1;
+            }
+        }
+
+        /// <summary>
+        /// Returns list of templates for unambiguous template group, otherwise empty list
+        /// </summary>
+        public IReadOnlyCollection<ITemplateMatchInfo> UnambiguousTemplateGroup
+        {
+            get
+            {
+                return HasUnambiguousTemplateGroup ? ExactMatchedTemplates : new List <ITemplateMatchInfo>();
+            }
+        }
+
+        /// <summary>
+        /// Returns true if all the templates in unambiguous group have templates in same language
+        /// </summary>
+        public bool AllTemplatesInUnambiguousTemplateGroupAreSameLanguage
+        {
+            get
+            {
+                if (UnambiguousTemplateGroup.Count == 0)
+                {
+                    return false;
+                }
+
+                if (UnambiguousTemplateGroup.Count == 1)
+                {
+                    return true;
+                }
+                    
+                HashSet<string> languagesFound = new HashSet<string>();
+                foreach (ITemplateMatchInfo template in UnambiguousTemplateGroup)
+                {
+                    string language;
+
+                    if (template.Info.Tags != null && template.Info.Tags.TryGetValue("language", out ICacheTag languageTag))
+                    {
+                        language = languageTag.ChoicesAndDescriptions.Keys.FirstOrDefault();
+                    }
+                    else
+                    {
+                        language = string.Empty;
+                    }
+
+                    if (!string.IsNullOrEmpty(language))
+                    {
+                        languagesFound.Add(language);
+                    }
+
+                    if (languagesFound.Count > 1)
+                    {
+                        return false;
+                    }
+                }
+                return true;
+            }
+        }
+
+
+        private IReadOnlyCollection<ITemplateMatchInfo> GetCoreMatchedTemplatesWithDisposition(Func<ITemplateMatchInfo, bool> filter)
```

Comment:
Do we really need this function? This is just calling `Where` on the data, which can already be done from the original call site.